### PR TITLE
Enable mkFit for initialStepPreSplitting with trackingMkFit modifier

### PR DESCRIPTION
**Pull request by makortel**
#### PR description:

This PR extends the mkFit setup (enabled with `trackingMkFit` modifier, X.7 workflows) to use mkFit in initialStepPreSplitting, as mentioned in [TRK POG meeting 2021-06-14](https://indico.cern.ch/event/1040448/#5-mkfit-status-report).

This PR also includes a technical fix to include a header of `PropagatorWithMaterial` instead of its source file (I have no idea how that ended up there).

#### PR validation:

Workflows 11634.0, 11634.7 run.

### CODE REVIEW

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
@cmsbuild, please abort

**Comment by makortel** [comment]
test parameters:
  - workflows = 11634.7

**Comment by makortel** [comment]
@cmsbuild, please test

File: RecoTracker/IterativeTracking/python/InitialStepPreSplitting_cff.py [hunk 1]
```diff
@@ -125,6 +125,45 @@
...
 
+from Configuration.ProcessModifiers.trackingMkFitInitialStepPreSplitting_cff import trackingMkFitInitialStepPreSplitting
+from RecoTracker.MkFit.mkFitGeometryESProducer_cfi import mkFitGeometryESProducer
+import RecoTracker.MkFit.mkFitSiPixelHitConverter_cfi as mkFitSiPixelHitConverter_cfi
+import RecoTracker.MkFit.mkFitSiStripHitConverter_cfi as mkFitSiStripHitConverter_cfi
+import RecoTracker.MkFit.mkFitEventOfHitsProducer_cfi as mkFitEventOfHitsProducer_cfi
+import RecoTracker.MkFit.mkFitSeedConverter_cfi as mkFitSeedConverter_cfi
+import RecoTracker.MkFit.mkFitIterationConfigESProducer_cfi as mkFitIterationConfigESProducer_cfi
+import RecoTracker.MkFit.mkFitProducer_cfi as mkFitProducer_cfi
+import RecoTracker.MkFit.mkFitOutputConverter_cfi as mkFitOutputConverter_cfi
```
Comment (jpata) [inline RecoTracker/IterativeTracking/python/InitialStepPreSplitting_cff.py:136]: sorry if I miss something, but should/could these be hidden from re-exporting (prefixed with `_`)?

File: RecoTracker/IterativeTracking/python/InitialStepPreSplitting_cff.py [hunk 2]
```diff
@@ -125,6 +125,45 @@
...
+# TODO: Split the MkHitConverter to two to avoid re-converting strip hits in initialStep
```
Comment (jpata) [inline RecoTracker/IterativeTracking/python/InitialStepPreSplitting_cff.py:137]: perhaps the TODOs inline can be unclear in terms of when/who should do it (given that this is a generic tracking file), so maybe these should be collected to an issue?

**Review by jpata — COMMENTED** [review]
@mtosi @vmariani please let us know if you have additional comments/concerns.

File: RecoTracker/IterativeTracking/python/InitialStepPreSplitting_cff.py (hunk 1 above)
Comment (makortel) [reply RecoTracker/IterativeTracking/python/InitialStepPreSplitting_cff.py:136]: Certainly.

File: RecoTracker/IterativeTracking/python/InitialStepPreSplitting_cff.py (hunk 2 above)
Comment (makortel) [reply RecoTracker/IterativeTracking/python/InitialStepPreSplitting_cff.py:137]: Whoops, that "TODO" was already addressed in #33802 and thus the comment should be removed.

File: RecoTracker/IterativeTracking/python/InitialStepPreSplitting_cff.py (hunk 1 above)
Comment (makortel) [reply RecoTracker/IterativeTracking/python/InitialStepPreSplitting_cff.py]: I changed only the `InitialStepPreSplitting_cff.py` that was modified in this PR. If you wish, I can change the other iterations where mkFit can be enabled accordingly as well.

**Comment by makortel** [comment]
@cmsbuild, please test

File: RecoTracker/IterativeTracking/python/InitialStepPreSplitting_cff.py (hunk 1 above)
Comment (jpata) [reply RecoTracker/IterativeTracking/python/InitialStepPreSplitting_cff.py]: thanks! maybe the other iterations can be changed in a (technical) follow-up?

**Comment by jpata** [comment]
+reconstruction
- mkFit in initialStepPreSplitting with a modifier
- the mkFit workflow 11634.7 runs
- no reco differences in standard workflows

**Comment by jfernan2** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**